### ISSUE TITLE
E2Rank Loss func to finetuning reranker

### ISSUE THREAD

**Issue by uchuuronin** [riakashyap/646_Project#30]
E2Rank loss: Layerwise cross-entropy + KL divergence. (pg3 of paper for reference)
    
    L_total = L_layerwise + L_divergence
    
Where:
    - L_layerwise = (1/|L|) * SUM_OF CE(y_l, y) over layers
    where y_l is the probability distribution at the l-th layer, 
    and y is the target one,
    
    - L_divergence = (1/(|L|-1)) * SUM_OF KL(y_l || y_final) over intermediate layers
    where y_l is the probability distribution at the l-th layer (student) 
    where y_l is the probability distribution at the l-th layer (teacher)

  Goal:
  1. Each layer should be able to rank positive doc highest (via CE)
  2. Intermediate layers should be consistent with final layer (via KL)
